### PR TITLE
its ready to pull 

### PR DESCRIPTION
**Pull request by whokarez**
- Devided the node list into 4 sections 2D,3D,Misc,All
- Added a full functional search bar with a Clear button
- fixed issue where u're not allowed to add the same node twice
- fixed issue where the catagory names where listed as nodes
- changed single click into a double click
- the drag n drop was enhanced but not finished yet so its still under development
--------------------------------
also edited the readme file to contain the update log , updated install tips and added all the Contributors names along with their discord & github accounts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,20 +2,59 @@
...
+
+
 
 ## How to install
+
 ### Manual install
 1. Download this repository as ZIP
 2. Extract the folder under your `res://addons` folder
 3. Go to `Project->ProjectSettings->Plugins` and enable `Drag Nodes to Scene`
+4. (make sure there is a folder named "Godot-DragNDropNodes-main" under the addons folder then put the files inside it ,
```
Comment (AnidemDex) [inline README.md:23]: We can avoid relying on full paths if we use relative paths in the plugin script

File: README.md [hunk 2]
```diff
@@ -2,20 +2,59 @@
...
-Subfolder name can be changed, plugin doesn't rely on the plugin folder path
+
+update log 
+---------------------------
+- Devided the node list into 4 sections 2D,3D,Misc,All 
+- Added a full functional search bar with a Clear button 
+- fixed issue where u're not allowed to add the same node twice 
+- fixed issue where the catagory names where listed as nodes 
+- changed single click into a double click 
+- the drag n drop was enhanced but not finished yet so its still under development
```
Comment (AnidemDex) [inline README.md:39]: I don't like the idea of merging incomplete stuff

File: README.md [hunk 3]
```diff
@@ -2,20 +2,59 @@
...
+discord: @anidemdex , github: @AnidemDex
+
+# jam abusheira 
+as designer and debugger 
+discord:@0whocarez0 , github: @0whokarez0 
+
+# dead bunny 
+as the git expert 
+discord: @ideadbunny , github:@ideadBunny
+
```
Comment (AnidemDex) [inline README.md]: I think it should be
```
# Contributors
## <Contributor name>
```
Or just a list under contributors title

File: plugin_script.gd [hunk 4]
```diff
@@ -1,85 +1,13 @@
...
-		if what == NOTIFICATION_READY:
-			generate_class_tree()
-	
-	func _init() -> void:
-		name = "Nodes"
-		item_selected.connect(_item_selected)
-
-var class_tree:ClassTree
+var class_tree
+const ClassTree = preload("res://addons/Godot-DragNDropNodes-main/class_tree.gd")  # Update the path if necessary
```
Comment (AnidemDex) [inline plugin_script.gd]: Maybe we can use `./class_tree.gd` to avoid relying in fullpath

File: class_tree.gd
Comment (AnidemDex) [inline class_tree.gd:1]: This script removes the drag data, doesn't it?

**Review by AnidemDex — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (whokarez) [reply README.md:39]: there is no incomplete code ( the folder for the drag n drop is separate on my pc i will deal with it later  i didnt even put any of it here , this code is solid and all the commits are full no half measures

File: README.md (hunk 1 above)
Comment (whokarez) [reply README.md:23]: i added the line ( any name will work ) it's just i wanted to emphasize the importance to have a subFolder before dumbing all the files (regardless of the folder name ) cuz some people might directly dumb them into addons\.. which happened when i sent it to my friend so .. just a friendly warning

File: README.md (hunk 3 above)
Comment (whokarez) [reply README.md]: sure i'll change it ,

File: plugin_script.gd (hunk 4 above)
Comment (0whokarez0) [reply plugin_script.gd:5]: i dunno how to do it , if you can do it go ahead 🤔

File: class_tree.gd
Comment (whokarez) [reply class_tree.gd:1]: actually yes but we're going to fix it in a later vesion with a much more superior one , i'll tell u about it later

File: README.md (hunk 3 above)
Comment (whokarez) [reply README.md]: i changed it to this ----> names with their github accounts


# Contributors
## <the Dex @AnidemDex>
## <whoKarez @0whokarez0>
## <deadBunny @ideadBunny>

Comment (AnidemDex) [reply README.md]: The <> was an indicator to replace 😅

File: plugin_script.gd (hunk 4 above)
Comment (whokarez) [reply plugin_script.gd]: wait now i've done it 👍
image

**Comment by whokarez** [comment]
finished all the changes @AnidemDex

**Pull request merged**